### PR TITLE
Stackfiles by feature

### PR DESCRIPTION
**Pull request by bquenin**
This PR groups services by feature into stack files:
- Core stack
- Logs stack
- Metrics stack
- Proxy stack

## How to test
```
$ ampmake clean build
$ amp cluster create
```
Check travis build

### CODE REVIEW

**Comment by neha-viswanathan** [comment]
Can you add `Deploying stack <stackname>` in the output for clarity?

**Comment by bquenin** [comment]
Done.

**Review by neha-viswanathan — APPROVED** [review]

File: cluster/ampagent/cmd/install.go [hunk 1]
```diff
@@ -148,10 +137,11 @@ func serviceDeploymentMode(c docker.APIClient, labelKey string, labelValue strin
 }
 
 // returns sorted list of yaml file pathnames
-func getStackFiles(path string, clusterMode map[string]string) ([]string, error) {
+func getStackFiles(path string, deploymentMode string) ([]string, error) {
 	if path == "" {
 		path = "./stacks"
 	}
+	path += "/" + deploymentMode
```
Comment (ndegory) [inline cluster/ampagent/cmd/install.go:144]: a consequence of the change is that all stack files are duplicated, even when there's no reason to duplicate it (the proxy stack is the same whatever the cluster mode). That means that if somebody forgot to report a change on both files, we'll miss the difference during the tests. Why not leveraging the _include_ syntax to load service definitions from a common file? That should also be done on the stacks that are different based on the cluster mode, but contains many services that are not different (in the core stack only etcd is dependent on the cluster mode).

Comment (bquenin) [reply cluster/ampagent/cmd/install.go:144]: Sounds good! Please open an issue.

**Pull request merged**